src/mirakuru/base.py

```diff
@@ -129,9 +130,26 @@ def __init__(
 
         self._uuid = str(uuid.uuid1())
 
+        # Python doesn't guarantee to call __del__ methods on process
+        # termination so here we catch INT, TERM and QUIT system signals
+        # and try to finally kill the subprocess if it's still running yet.
+
+        # We create a weak reference to allow garbage collection in case
+        # executor is deleted before the current process termination.
+        self_weak = weakref.ref(self)
+
+        def on_terminate(signum, frame):
+            executor = self_weak()
+            if executor:
+                executor.kill()
+
+        signal.signal(signal.SIGINT, on_terminate)
```

Comment:
I will check if [atexit](https://docs.python.org/2/library/atexit.html) catches INT, TERM and QUIT signals then.